### PR TITLE
[launcher] Update the token refresh logic

### PR DESCRIPTION
**Pull request by jkl73**
Instead of directly overwriting the token file (risking partial read race condition). Creating a tmp file first and then rename the tmp file to the token file.

### CODE REVIEW

File: launcher/container_runner.go [hunk 1]
```diff
@@ -356,9 +359,29 @@ func (r *ContainerRunner) refreshToken(ctx context.Context) (time.Duration, erro
...
-	if err = os.WriteFile(filepath, token, 0644); err != nil {
-		return 0, fmt.Errorf("failed to write token to container mount source point: %v", err)
+	// Write to a temp file first.
+	tmpTokenPath := path.Join(hostTokenPath, tokenFileTmp)
+	if err = os.WriteFile(tmpTokenPath, token, 0644); err != nil {
+		return 0, fmt.Errorf("failed to write a tmp token file: %v", err)
+	}
+
+	// Rename the temp file to the token file (to avoid race conditions).
+	if err = os.Rename(tmpTokenPath, path.Join(hostTokenPath, tokenFile)); err != nil {
```
Comment (alexmwu) [inline launcher/container_runner.go:369]: as @KeithMoyer suggested you could just move tokenFileTmp to tokenFile

**Review by alexmwu — COMMENTED** [review]
Have you thought about how we might test this?

**Comment by jkl73** [comment]
> Have you thought about how we might test this?

Basic test will make sure the token is accessible (initial token), I plan to run a VM for an extended period of time and test manually.

**Comment by jkl73** [comment]
> > Have you thought about how we might test this?
> 
> Basic test will make sure the token is accessible (initial token), I plan to run a VM for an extended period of time and test manually (for refreshed token)

Tested in a VM, is able to read the refreshed token:
[  102.334098] cs_container_launcher[686]: claim exp:  2023-07-05 23:11:55 +0000 UTC
[  102.344224] cs_container_launcher[686]: claim isa:  2023-07-05 22:11:55 +0000 UTC
[ 3099.723244] cs_container_launcher[686]: refreshing attestation verifier OIDC token
[ 3702.249397] cs_container_launcher[686]: time:  2023-07-05 23:11:56.043983177 +0000 UTC m=+3600.010760865
...
[ 3702.290146] cs_container_launcher[686]: claim exp:  2023-07-06 00:01:54 +0000 UTC
[ 3702.300115] cs_container_launcher[686]: claim isa:  2023-07-05 23:01:54 +0000 UTC

**Review by alexmwu — APPROVED** [review]

**Pull request merged**